File: Go/Persona.go
type Persona struct {
	id     int
	edad   int
	genero bool
}


func NuevaPersona(id int, edad int, genero bool) *Persona {
	return &Persona{id: id, edad: edad, genero: genero}
}


func (p *Persona) GetId() int {
	return p.id
}

func (p *Persona) GetEdad() int {
	return p.edad
}

func (p *Persona) GetGenero() bool {
	return p.genero
}


func (p *Persona) SetId(nuevoId int) {
	p.id = nuevoId
}

func (p *Persona) SetEdad(nuevaEdad int) {
	p.edad = nuevaEdad
}


func (p *Persona) MostrarDatos() {
	var g string
	if p.genero {
		g = "Masculino"
	} else {
		g = "Femenino"
	}
	fmt.Println("ID:", p.id, "| Edad:", p.edad, "| Género:", g)
}


func (p *Persona) Resta(a int, b int) int {
	resultado := a - b
	return resultado
}


func (p *Persona) ToJSON() string {
	return fmt.Sprintf("id: %d, edad: %d, genero: %t", p.id, p.edad, p.genero)
}

File: java/Persona.java
public abstract class Persona {

    public int id;
    public int edad;
    public boolean genero;

    
    public Persona(int id, int edad, boolean genero) {
        this.id = id;
        this.edad = edad;
        this.genero = genero;
    }

  
    public int getId() {
        return this.id;
    }

    public int getEdad() {
        return this.edad;
    }

    public boolean getGenero() {
        return this.genero;
    }

    public void setId(int nuevoId) {
        this.id = nuevoId;
    }

    public void setEdad(int nuevaEdad) {
        this.edad = nuevaEdad;
    }

    public void mostrarDatos() {
        System.out.println("ID: " + id + " | Edad: " + edad + " | Género: " + (genero ? "Masculino" : "Femenino"));
    }
        public int resta (int a, int b) {
            int resultado = a - b;
            return  resultado;  

    }
    public String toJSON() {
        return "id: " + id + "edad: " + edad + "genero: " + genero; 

}
}


